server/src/main/java/io/crate/execution/engine/collect/files/LocalFsFileInput.java

```diff
@@ -28,53 +28,68 @@
 import java.io.InputStream;
 import java.net.URI;
 import java.nio.file.AccessDeniedException;
+import java.nio.file.FileSystemLoopException;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.util.ArrayList;
+import java.util.EnumSet;
 import java.util.List;
 import java.util.function.Predicate;
 
+import static java.nio.file.FileVisitOption.FOLLOW_LINKS;
+
 public class LocalFsFileInput implements FileInput {
 
     @Override
-    public List<URI> listUris(final URI fileUri, final Predicate<URI> uriPredicate) throws IOException {
+    public List<URI> listUris(final URI fileUri, final URI preGlobUri, final Predicate<URI> uriPredicate) throws IOException {
         assert fileUri != null : "fileUri must not be null";
+        assert preGlobUri != null : "preGlobUri must not be null";
         assert uriPredicate != null : "uriPredicate must not be null";
 
-        Path path = Paths.get(fileUri);
-        if (!Files.isDirectory(path)) {
-            path = path.getParent();
-            if (path == null) {
+        Path preGlobPath = Paths.get(preGlobUri);
+        if (!Files.isDirectory(preGlobPath)) {
+            preGlobPath = preGlobPath.getParent();
+            if (preGlobPath == null) {
                 return List.of();
             }
         }
-        if (Files.notExists(path)) {
+        if (Files.notExists(preGlobPath)) {
             return List.of();
         }
-
+        final int FILE_URI_DEPTH = countOccurrences(fileUri.toString(), '/');
+        final int MAX_DEPTH = FILE_URI_DEPTH - countOccurrences(preGlobUri.toString(), '/') + 1;
```

Comment:
```suggestion
        final int fileURIDepth = countOccurrences(fileUri.toString(), '/');
        final int maxDepth = fileURIDepth - countOccurrences(preGlobUri.toString(), '/') + 1;
```

We use all upper case only for constants (`final static` globals)